_posts/2022-12-16-aave-rebasing.md

```diff
@@ -0,0 +1,332 @@
+Aave's ERC20's are called [aTokens](https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/tokenization/AToken.sol).
+If you deposited [DAI](https://polygonscan.com/address/0x8f3cf7ad23cd3cadbd9735aff958023239c6a063) on Polygon, you would receive [aPolDAI](https://polygonscan.com/address/0x82E64f49Ed5EC1bC6e43DAD4FC8Af9bb3A2312EE) back.
+[Here](https://docs.aave.com/developers/deployed-contracts/v3-mainnet/polygon#tokens) is a list of other commonly-issued aTokens.
+
+aTokens are _receipt tokens_: they represent a claim on deposited assets.
+When you want to [withdraw](https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/pool/Pool.sol#L197-L217) your deposit and claim its accumulated interest, your aTokens are [burned](https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/libraries/logic/SupplyLogic.sol#L131-L136).
+
+One thing that makes Aave's aTokens interesting is that their balance programmatically increases over time.
+If you have (say) 100
+[aPolWETH](https://polygonscan.com/address/0xe50fA9b3c56FfB159cB0FCA61F5c9D750e8128c8) today, you will have 102 aPolWETH in a year from now.
+That's without making any new deposits or interacting with the protocol or anything.
+It's [built
+into](https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/tokenization/AToken.sol#L131-L139) the aToken contract.
+
+This is called _rebasing_.
+Said another way, aTokens are _rebase tokens_.
+
+Aave has a really interesting implementation of rebasing -- one which we believe is not well understood.
+One of the things we want to do in this post is explain how it works.
+
+#### Liquidity Indexes
+
+A key concept in understanding aToken rebasing is the notion of a __liquidity index__.
+It's defined in their [v2
+whitepaper](https://raw.githubusercontent.com/aave/protocol-v2/master/aave-v2-whitepaper.pdf) as follows:
+
+> Interest cumulated by the reserve during the time interval ∆T, updated
+> whenever a borrow, deposit, repay,
+> redeem, swap, liquidation event occurs.
+
+In other words, __the liquidity index for a certain asset is the total amount of interest earned by
+deposits of that asset into Aave__ -- expressed as a percentage scaled up by a
+[ray](https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/libraries/math/WadRayMath.sol#L17).
+
+Let's unpack that for a second, because it's a mouthful and this is really important.
+
+aPolyDAI currently has a liquidity index of `1010832786421347125844242759`, or
+1.083%. This means that _in aggregate_, over the entire lifetime of Aave, DAI deposits have earned 1.083%.
+So if you were the very first person to deposit DAI into
+Aave and you haven't touched it since, your deposit would today be worth 1.083% more.
```

Comment:
Haven't they earned 0.083%? Because `1e27 * (1 + 0.083) = 1.083e27`

```suggestion
aPolyDAI currently has a liquidity index of `1010832786421347125844242759`, or
1.083. This means that _in aggregate_, over the entire lifetime of Aave, DAI deposits have earned 1.083 - 1 = 0.083%.
So if you were the very first person to deposit DAI into
Aave and you haven't touched it since, your deposit would today be worth 0.083% more.
```

IIUC the index itself isn't a percentage, it's `1 + interest`, which becomes `1e27 + interest` as a ray, so there's a few other places below where we'd want to drop the percentage sign

Reply:
I think you might be missing a couple digits there.

> the index itself isn't a percentage, it's 1 + interest

Yup, that's right. But the index is `1010832786421347125844242759`, i.e. `1.01083`, if you subtract `1` you end up with `0.01083`, or `1.083%`

Reply:
I'd mark this resolved, but seems I don't have permission to do that here